sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java

```diff
@@ -46,6 +51,10 @@ public static void main(String[] args) throws IOException {
                     .setLogLevel(parsedArguments.getHttpLogDetailLevel()))
             .buildClient();
 
+        // This mapper gets used to deserialize a digital twin that has a date time within a property metadata, so it
+        // needs to have this module in order to correctly deserialize that date time
+        mapper.registerModule(new JavaTimeModule());
```

Comment:
This is the module that was missing from our default mapper in the digital twins client. We get this module for free when we construct the JacksonAdapter and call .serializer() on that jacksonAdapter. We need to explicitly use it here, though